distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type.enso

```diff
@@ -0,0 +1,243 @@
+            Error.throw (Invalid_Value_Type.Error Value_Type.Char value_type related_column)
+
+    ## ADVANCED
+       UNSTABLE
+       Checks if the provided value type is a boolean type and runs the
+       following action or reports a type error.
+    expect_boolean : Value_Type -> Any -> Any ! Invalid_Value_Type
+    expect_boolean value_type ~action = case value_type of
+        Value_Type.Boolean -> action
+        _ -> Error.throw (Invalid_Value_Type.Error Value_Type.Boolean value_type)
+
+    ## UNSTABLE
+       Checks if the `Value_Type` represents a boolean type.
+    is_boolean : Boolean
+    is_boolean self = case self of
+        Value_Type.Boolean -> True
+        _ -> False
+
+    ## UNSTABLE
+       Checks if the `Value_Type` represents a floating-point number type.
+    is_floating_point : Boolean
+    is_floating_point self = case self of
+        Value_Type.Float _ -> True
+        _ -> False
+
+    ## UNSTABLE
+       Checks if the `Value_Type` represents a text type.
+    is_text : Boolean
+    is_text self = case self of
+        Value_Type.Char _ _ -> True
+        _ -> False
+
+    ## UNSTABLE
+       Checks if the `Value_Type` represents any numeric type - integer,
+       floating point or decimal.
+    is_numeric : Boolean
+    is_numeric self = case self of
+        Value_Type.Integer _ -> True
+        Value_Type.Float _   -> True
+        Value_Type.Decimal _ _ -> True
+        _ -> False
+
+    ## UNSTABLE
+       Checks if the `Value_Type` represents an integer type.
+    is_integer : Boolean
+    is_integer self = case self of
+        Value_Type.Integer _ -> True
+        _ -> False
+
+    ## Provides a text representation of the `Value_Type` meant for
+       displaying to the user.
+    to_display_text : Text
+    to_display_text self = case self of
+        Value_Type.Boolean -> "Boolean"
+        Value_Type.Byte -> "Byte"
+        Value_Type.Integer size -> "Integer (" + size.to_text + ")"
+        Value_Type.Float size -> "Float (" + size.to_text + ")"
+        Value_Type.Decimal precision scale -> "Decimal (precision=" + precision.to_text + ", scale=" + scale.to_text + ")"
+        Value_Type.Char size variable_length -> "Char (max_size=" + size.to_text + ", variable_length=" + variable_length.to_text + ")"
+        Value_Type.Date -> "Date"
+        Value_Type.Date_Time with_timezone -> "Date_Time (with_timezone=" + with_timezone.to_text + ")"
+        Value_Type.Time -> "Time"
+        Value_Type.Binary size variable_length -> "Binary (max_size=" + size.to_text + " bytes, variable_length=" + variable_length.to_text + ")"
+        Value_Type.Unsupported_Data_Type type_name _ -> case type_name of
+            Nothing  -> "Unsupported_Data_Type"
+            _ : Text -> "Unsupported_Data_Type (" + type_name + ")"
+        Value_Type.Mixed -> "Mixed"
+
+    ## PRIVATE
+       Provides a JS object representation for use in visualizations.
+    to_js_object : JS_Object
+    to_js_object self =
```

Comment:
Shouldn't this be in `Standard.Visualization` library?

Reply:
I think you are right! I will try to move it.

Reply:
Apparently, moving it causes trouble because then I need to ensure that the extension method is imported. And it does not work in as predictable way as I'd like it to.

So I think I'd keep it here for now to avoid the visualization not working unexpectedly.

cc: @jdunkerley where do you think we should place such methods? I tried moving it to `Standard.Visualization.Helpers`, but then it was not registering properly in the `Visualization_Tests`.

Reply:
The `to_js_object` should be here. I am going to amend the Table viz to add display text itself at which point that should be somewhere else. For now this works.